### PR TITLE
Fix: DO-2027 none in numeric input shows 0

### PR DESCRIPTION
**Pull request by patricia-causalens**
## Motivation and Context

Claudia found that when starting NumericInput with Variable(), it displayed 0 instead of appearing empty

## Implementation Description

This was because Number(null) returns 0and not null. Although the variable contained the correct value the display did not match the same behaviour.

How the component behaves:
If starting with Variable() -> empty display
If Variable has a number -> displays the number
If Variable contains a string that can be converted to a number -> displays the number
If Variable contains something that cannot be converted to a number -> empty display

I checked that the above is followed for both initialising the Variable with those and by updating the Variable value through an action

## Any new dependencies Introduced

None

## How Has This Been Tested?

Locally for all the scenarios covered above

## PR Checklist:

- [x] I have implemented all requirements? (see JIRA, project documentation).
- [x] I am not affecting someone else's work, If I am, they are included as a reviewer.
- [ ] I have added relevant tests (unit, integration or regression).
- [ ] I have added comments to all the bits that are hard to follow.
- [ ] I have added/updated Documentation.
- [x] I have updated the appropriate changelog with a line for my changes.

### CODE REVIEW

File: packages/dara-components/changelog.md [hunk 1]
```diff
@@ -2,6 +2,10 @@
 title: Changelog
 ---
 
+## NEXT
+
+-   Fixed an issue where `Input` of type number displayed zero istead of null when null value was set.
```
Comment (krzysztof-causalens) [inline packages/dara-components/changelog.md:7]: typo `istead` -> `instead`

File: packages/dara-components/js/common/input/input.tsx [hunk 2]
```diff
@@ -24,6 +24,9 @@ interface InputProps extends FormComponentProps {
 const StyledInput = injectCss(UiInput);
 const StyledNumericInput = injectCss(UiNumericInput);
 
+function getNumericValue(value: string | number): number {
+    return value == null || Number.isNaN(+value) ? null : +value;
```
Comment (krzysztof-causalens) [inline packages/dara-components/js/common/input/input.tsx:28]: What's the purpose of using `+` rather than `Number()` here?

**Review by krzysztof-causalens — COMMENTED** [review]
Looks good, quick question

File: packages/dara-components/js/common/input/input.tsx (hunk 2 above)
Comment (patricia-causalens) [reply packages/dara-components/js/common/input/input.tsx:28]: They both behave the same way here, I learned about it the other day and had just been looking for an excuse to try it.

File: packages/dara-components/changelog.md (hunk 1 above)
Comment (patricia-causalens) [reply packages/dara-components/changelog.md]: oops fixed

**Review by claudiawestby — APPROVED** [review]
thanks so much :)

File: packages/dara-components/js/common/input/input.tsx (hunk 2 above)
Comment (krzysztof-causalens) [reply packages/dara-components/js/common/input/input.tsx:28]: I would argue `Number` is more explicit, with `+` for example I had to think twice whether there wouldn't be some edge case with negative numbers.

Comment (patricia-causalens) [reply packages/dara-components/js/common/input/input.tsx]: That is fair, I have switched to use `Number`

**Review by krzysztof-causalens — APPROVED** [review]

**Pull request merged**